ast/types.ts

```diff
@@ -0,0 +1,26 @@
+export type AST = Array<ClassNode>;
+
+export type Modifiers = Array<string>;
+
+export type ClassNode = {
+  type: "class";
+  modifiers: Modifiers;
+  name: string;
+  body: {
+    methods: Array<MethodNode>;
+  }
+};
+
+export type Param = {
+  typeName: string;
+  argName: string;
+};
+
+export type MethodNode = {
+  type: "method";
+  modifiers: Modifiers;
+  name: string;
+  returnType: string;
+  params: Array<Param>;
+  body: Array<string>;
+};
```

Comment:
```suggestion
export type CompilationUnit = Array<NormalClassDeclaration>

export interface NormalClassDeclaration {
    classModifier?: ClassModifier;
    typeIdentifier: Identifier;
    classBody: Array<MethodDeclaration>;
}

export interface MethodDeclaration {
    methodModifier?: Array<MethodModifier>;
    result: Result;
    identifier: Identifier;
    formalParameterList: Array<FormalParameter>;
}

export interface FormalParameter {
    unannType: UnannReferenceType;
    variableDeclaratorId: Identifier;
}

export type ClassModifier = 
    | "public" 
    | "protected"
    | "private" 
    | "abstract" 
    | "static" 
    | "final"
    | "sealed"
    | "non-sealed"
    | "strictfp";

export type MethodModifier = 
    | "public" 
    | "protected"
    | "private" 
    | "abstract" 
    | "static" 
    | "final"
    | "synchronized"
    | "native"
    | "strictfp";

export type Result = "void";

export type UnannReferenceType = "String[]";

export type Identifier = string;

```

Reply:
Perhaps following the terminologies in [JLS](https://docs.oracle.com/javase/specs/jls/se20/html/jls-19.html) would ease future extension?

Reply:
Ya, I agree and have made the relevant changes.